### ISSUE TITLE
`event.keycode` is incorrect on function keys

### ISSUE THREAD

**Issue by matan-h** [LadybirdBrowser/ladybird#974]
This is discovered through the #972 issue. 
On a website like this:
```html
<script>
    document.addEventListener("keydown", event =>console.log(event.key,event.keyCode))
</script>
```

when the user presses the function keys, the `event.key` is the correct name, but `event.keyCode` equal `<the correct value>-85` (e.g. `F12` is supposed to be `123` is `38`.) 
other function key get other margins:

key    | Correct keyCode | ladybird keyCode
|--------|--------|--------|
| F1 | 122 | 27 |
| NumLock | 144 | 25 |
| ScrollLock | 145 | 26 | 
| Delete | 46 | 6 |
| Insert | 45 | 5 |

My intuitive thought was to implement the function keys in [`KeyboardEvent.cpp`](https://github.com/LadybirdBrowser/ladybird/blob/1e7b17f15024d0618f1a1e17e3e927856febde28/Userland/Libraries/LibWeb/UIEvents/KeyboardEvent.cpp#L54) because capsLock seems to be implemented there, but I don't understand c++ that much.

<details>
<summary>Version</summary>

```yml
Version: 1.0
Arch:  x86_64
Operating System:  Linux <Fedora KDE 40> [Wayland]
Git hash <git rev-parse HEAD>: 1e7b17f15024d0618f1a1e17e3e927856febde28
```
</details>

**Comment by circl-lastname** [comment]
So this is the same issue as #972

**Comment by matan-h** [comment]
Not really, #972 is specifically about implementing F12 to close the inspector window.

I just found this issue when I tried to implement that in the inspector javascript.

**Comment by Arhcout** [comment]
I looked through the definition of the key codes and the F keys are misplaced:
`enum KeyCode : u8 {...`
`__ENUMERATE_KEY_CODE(F1, "F1") // 27th element of the enum`
`...}`
It's the same for other keys.
A simple fix should be to move the keys definitions in the proper place so that they have the right code assigned to them.

**Comment by circl-lastname** [comment]
Ah, I see

**Comment by matan-h** [comment]
@Arhcout I'm trying it now, and it seems like the codes have [too many](https://www.toptal.com/developers/keycode/table) blank spaces to continue using `__ENUMERATE_KEY_CODE` (i.e. it starts with 3,8,9,12). What is the best way to create this kind of struct in c++ ?

**Comment by circl-lastname** [comment]
It depends if you folks are looking at the keycode definitions used for LibWeb pages VS the internal libraries. KeyboardEvent.cpp appears to translate the platform's KeyCode to the one expected by a JS application, therefore Function keys should be added there, not the KeyCode enum rearranged.

**Comment by Arhcout** [comment]
I defined all of the keycodes and I will make a pull request.

**Comment by AtkinsSJ** [comment]
Fixed in #989
